### PR TITLE
Template Update

### PR DESCRIPTION
**Pull request by mephenor**
Decided to factor out some basic changes affecting both of the branches I was working on.

This PR includes:

- template updates
- version bumps
- pydantic changes, i.e. `json()` -> `model_dump_json()`, etc.
- updates to reset_fixture, i.e. `kafka.clean_topics()` and added `scope="module"` to edge case tests

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ repos:
       - id: debug-statements
       - id: debug-statements
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.1.3
+    rev: v0.1.14
     hooks:
       - id: ruff
         args: [--fix, --exit-non-zero-on-fix]
```
Comment (TheByronHimes) [inline .pre-commit-config.yaml:54]:
```suggestion
        args: [--fix, --exit-non-zero-on-fix]
      - id: ruff-formatter
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -51,7 +51,7 @@ repos:
       - id: debug-statements
       - id: debug-statements
```
Comment (TheByronHimes) [inline .pre-commit-config.yaml]: Nuke black and the three trailing debug-statements too

**Review by TheByronHimes — CHANGES_REQUESTED** [review]

File: .template/deprecated_files_ignore.txt
Comment (TheByronHimes) [inline .template/deprecated_files_ignore.txt:1]: This file probably doesn't need to be here if it's not being used.

File: .template/static_files_ignore.txt
Comment (TheByronHimes) [inline .template/static_files_ignore.txt:1]: Same as `deprecated_files_ignore.txt`

File: .template/mandatory_files_ignore.txt
Comment (TheByronHimes) [inline .template/mandatory_files_ignore.txt:1]: Same as `deprecated_files_ignore.txt`

**Review by TheByronHimes — DISMISSED** [review]
Looks good, suggested deletions are optional

**Review by TheByronHimes — CHANGES_REQUESTED** [review]
I just noticed that the top-level Dockerfile needs to be updated. Replace lines 35-36 with: 
```dockerfile
COPY --from=builder /service/lock/requirements.txt /service
RUN pip install --no-deps -r requirements.txt
RUN rm requirements.txt
COPY --from=builder /service/dist/ /service
RUN pip install --no-deps *.whl
RUN rm *.whl
```

**Comment by mephenor** [comment]
I can add that now, but that's already part of one of the coming PR's building on top of it

**Review by TheByronHimes — APPROVED** [review]
Sounds good then.
